packages/features/bookings/lib/service/RegularBookingService.ts

```diff
@@ -2243,32 +2288,36 @@ async function handler(
 
   const metadata = videoCallUrl
     ? {
-        videoCallUrl: getVideoCallUrlFromCalEvent(evt) || videoCallUrl,
-      }
+      videoCallUrl: getVideoCallUrlFromCalEvent(evt) || videoCallUrl,
+    }
     : undefined;
 
-  const bookingFlowConfig = {
+  const bookingCreatedPayload = buildBookingCreatedPayload({
+    booking,
+    organizerUserId: organizerUser.id,
+    // FIXME: It looks like hasHashedBookingLink is set to true based on the value of hashedLink when sending the request. So, technically we could remove hasHashedBookingLink usage completely
+    hashedLink: hasHashedBookingLink ? reqBody.hashedLink ?? null : null,
     isDryRun,
-  };
+    organizationId: eventOrganizationId,
+  });
 
-  const bookingCreatedPayload = {
-    config: bookingFlowConfig,
-    bookingFormData: {
-      // FIXME: It looks like hasHashedBookingLink is set to true based on the value of hashedLink when sending the request. So, technically we could remove hasHashedBookingLink usage completely
-      hashedLink: hasHashedBookingLink ? reqBody.hashedLink ?? null : null,
-    },
+  const bookingRescheduledPayload: BookingRescheduledPayload = {
+    ...bookingCreatedPayload,
+    oldBooking: originalRescheduledBooking ? {
+      startTime: originalRescheduledBooking.startTime,
+      endTime: originalRescheduledBooking.endTime,
+    } : undefined,
   };
 
-  // Add more fields here when needed
-  const bookingRescheduledPayload = bookingCreatedPayload;
-
   const bookingEventHandler = deps.bookingEventHandler;
 
   // TODO: Incrementally move all stuff that happens after a booking is created to these handlers
   if (originalRescheduledBooking) {
     await bookingEventHandler.onBookingRescheduled(bookingRescheduledPayload);
   } else {
-    await bookingEventHandler.onBookingCreated(bookingCreatedPayload);
+    // TODO: We need to check session in booking flow and accordingly create USER actor if applicable.
+    const auditActor = makeGuestActor({ email: bookerEmail, name: fullName });
```

Comment:
Likewise should this be handled in another service? Maybe the one on the line below? I want to avoid needing people to call these specific functions in different features.

Reply:
I am not so sure about this as well. If you look at this definition, it isn't doing much at all.
My reasoning was that everywhere one would need to define these actors,  and it would be a lot of places. Either we pass the as a POJO or we have a well defined utility fn to create it, so that it is difficult to make mistakes

<img width="581" height="328" alt="Screenshot 2025-12-06 at 8 31 34 AM" src="https://github.com/user-attachments/assets/d253b191-cd4e-4b65-bc9b-bb6f83e9b3cf" />